pkg/retainer/txretainer/testsuite_test.go

```diff
@@ -0,0 +1,85 @@
+package txretainer_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gorm.io/gorm"
+
+	"github.com/iotaledger/hive.go/db"
+	"github.com/iotaledger/hive.go/log"
+	"github.com/iotaledger/hive.go/runtime/workerpool"
+	"github.com/iotaledger/hive.go/sql"
+	"github.com/iotaledger/iota-core/pkg/retainer/txretainer"
+	"github.com/iotaledger/iota-core/pkg/storage/prunablesql"
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+type txRetainerTestsuite struct {
+	T          *testing.T
+	currentDB  *prunablesql.LockableGormDB
+	TxRetainer *txretainer.TransactionRetainer
+
+	latestCommitedSlot iotago.SlotIndex
+	finalizedSlot      iotago.SlotIndex
+}
+
+func newTestSuite(t *testing.T) *txRetainerTestsuite {
+	dbParams := sql.DatabaseParameters{
+		Engine:   db.EngineSQLite,
+		Path:     t.TempDir(),
+		Filename: "test.db",
```

Comment:
```suggestion
		Filename: "txRetainerTestSQLite.db",
```

In case we ever want to find the test file locally, it's better to have a more descriptive name. For this to make any sense, my assumption is that the file is not immediately deleted after the test.